### ISSUE TITLE
Bug: selector-no-type - Ignore compounded allows types compounded by sibling selector

### ISSUE THREAD

**Issue by hankthewhale** [stylelint/stylelint#2256]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
In the [docs](http://stylelint.io/user-guide/rules/selector-no-type/), ignoring compounded seems to describe the ideal ignore pattern. `h1.class`. 

However, it also allows two element types to be compounded by the direct sibling selector `input[type='checkbox']:checked + label // does not fail`.

In this case, you can still style element types and sneak past the lint rule.

> Which rule, if any, is this issue related to?

selector-no-type

> What CSS is needed to reproduce this issue?

```css
input[type='checkbox']:checked + label {
  background: black;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
     "selector-no-type": [true, {
        "ignore": ["compounded"]
      }]
  }
}
```

> Which version of stylelint are you using?

`7.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`npm run stylelint`
`stylelint \"client/**/*.css\"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Warnings to return

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings are returned. However, warnings are returned when removing `ignore: ['compounded']` from the rule option

**Comment by jeddy3** [comment]
@hankthewhale Thanks for using the template.

I can't replicate this I'm afraid.

Given:

```css
input[type='checkbox']:checked + label {
  background: black;
}
```

The following config:

```json
{
  "rules": {
     "selector-no-type": [true, {
        "ignore": ["compounded"]
      }]
  }
}
```

Gives the following output:

```console
1:34 error Unexpected type selector (selector-no-type)
```

Whereas the following config:

```json
{
  "rules": {
     "selector-no-type": true
  }
}
```

Gives the following output:

```console
1:1 error Unexpected type selector (selector-no-type)
1:34 error Unexpected type selector (selector-no-type)
```

It looks like each _compounded/simple_ selector within a _complex_ selector is being correctly checked individually.

Can you create a reproducible repo for us so we can get to the bottom of this please?

**Comment by hankthewhale** [comment]
@jeddy3. Interesting. Yeah, I'll try to get an example repo up today. Thanks for looking into it.

**Comment by hankthewhale** [comment]
@jeddy3 Try it with the following rule

```
"selector-no-type": [true, {
  "ignore": ["compounded", "descendant"]
}]
```

Seems like ignoring descendants is maybe not catching the direct sibling selector. Going to dig into the stylelint code now.

**Comment by hankthewhale** [comment]
Looks like it might be an issue with [postcss-resolve-nested-selector](https://github.com/davidtheclark/postcss-resolve-nested-selector)

**Comment by jeddy3** [comment]
> I stupidly omitted the "descendant" option when writing the original bug report thinking that it had nothing to do with it. 

No worries :)

The issue you describe was fixed in https://github.com/stylelint/stylelint/pull/2200 and will be released in [`8.0.0`](https://github.com/stylelint/stylelint/milestone/23).

Going to close as a duplicate. Feel free to reopen if that's not it.

**Comment by hankthewhale** [comment]
Awesome! Was just starting fix this. Didn't realize it was already taken care of. 🎉 

Going to point at the `v8` branch until release as this is the single most important rule for us since we're using css-modules.

**Comment by hankthewhale** [comment]
@jeddy3 Added a (hopefully) helpful detail to the issue template to prevent duplicate bug reports in the future. https://github.com/stylelint/stylelint/pull/2258
